src/parsers.rs

```diff
@@ -210,11 +245,111 @@ fn timezone_utc(i: &[u8]) -> IResult<&[u8], (i32, i32)> {
     map(tag(b"Z"), |_| (0, 0))(i)
 }
 
-// Full ISO8601
+// Full ISO8601 datetime
 pub fn parse_datetime(i: &[u8]) -> IResult<&[u8], DateTime> {
     let (i, d) = parse_date(i)?;
     let (i, _) = tag(b"T")(i)?;
     let (i, t) = parse_time(i)?;
 
     Ok((i, DateTime { date: d, time: t }))
 }
+
+// DURATION
+
+// Y[YYY]
+fn duration_year(i: &[u8]) -> IResult<&[u8], u32> {
+    m_to_n_digit_in_range(i, 1, 4, 0..=9999)
```

Comment:
Hm, do we really need to limit this to 9999 years? A longer duration seems totally valid.

Reply:
Removed the 9999 limitation.